roles/ovb/bmc/tasks/main.yml

```diff
@@ -0,0 +1,8 @@
+# Setup BMC node
+- include: packages.yml
+- include: deploy_files.yml
+- include: os-net-config.yml
+
+# Deploy service file for each node in 'overcloud_nodes' group
+- include: openstack-bmc.yml
+  with_dict: "{{ provisioner.nodes }}"
```

Comment:
why is this outside the play ?
you are gathering the cloud details for nothing for every component no?
Why not use "with_subelements[1]" instead? since you know the values you're looking for, this is your solution

[1] - http://docs.ansible.com/ansible/playbooks_loops.html#looping-over-subelements